my_compassion/controllers/my2_event_banner.py

```diff
@@ -0,0 +1,72 @@
+##############################################################################
+#
+#    Copyright (C) 2025 Compassion CH (http://www.compassion.ch)
+#    @author: Elias Keller <ekeller@compassion.ch>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+from odoo import fields, http
+from odoo.http import request
+
+
+def _normalize_current_page_route(route: str) -> str:
+    if not route:
+        return "/"
+    route = route.split("?", 1)[0].split("#", 1)[0]
+    return route
+
+
+class MyCompassionEventBannerController(http.Controller):
+    @http.route("/my2/active-event-banners", type="json", auth="public", website=True)
+    def get_active_banner(self, current_page_route=None, limit=None, **kw):
+        current_page_route = _normalize_current_page_route(current_page_route)
+
+        now = fields.Datetime.now()
+        domain = [
+            ("is_active", "=", True),
+            ("start_date", "<=", now),
+            "|",
+            ("end_date", "=", False),
+            ("end_date", ">=", now),
+            ("target_route_ids.path", "=", current_page_route),
+        ]
+
+        banners = (
+            request.env["event.banner"]
+            .sudo()
+            .search(
+                domain,
+                order="start_date desc, id desc",
+                limit=limit or None,
+            )
+        )
+
+        rendered_banners = []
+
+        if not banners:
+            return rendered_banners
+
+        for banner in banners:
+            values = {
+                "id": banner.id,
+                "title": banner.title,
+                "description": banner.description,
+                "text_color": banner.text_color,
+                "start_date": banner.start_date,
+                "end_date": banner.end_date,
+                "is_active": banner.is_active,
+                "pictogram": banner.pictogram,
+                "pictogram_color": banner.pictogram_color,
+                "background_color": banner.background_color,
+                "button_text": banner.button_text,
+                "button_text_color": banner.button_text_color,
+                "button_background_color": banner.button_background_color,
+                "close_icon_color": banner.close_icon_color,
+                "button_action_url": banner.button_action_url,
+            }
+            html = request.env["ir.ui.view"]._render_template(
+                "theme_compassion_2025.EventBannerComponent", values
+            )
```

Comment:
Why not simply passing the whole banner record to the template and access the properties inside the template?
```suggestion
            html = request.env["ir.ui.view"]._render_template(
                "theme_compassion_2025.EventBannerComponent", {"banner": banner}
            )
```